app/views/bulk_grant_permission_sets/new.html.erb

```diff
@@ -1,25 +1,14 @@
-<% content_for :title, "Grant permissions to all users" %>
-
-<div class="page-title">
-  <h1>Grant permissions to all users</h1>
-</div>
-
-<div class="well">
-  <%= form_for @bulk_grant_permission_set do |f| %>
-    <h2>Permissions to grant</h2>
-
-    <% if @bulk_grant_permission_set.errors.count > 0 %>
-      <div class="alert alert-danger">
-        <ul>
-          <% @bulk_grant_permission_set.errors.full_messages.each do |message| %>
-            <%= content_tag :li, message %>
-          <% end %>
-        </ul>
-      </div>
-    <% end %>
-
-    <%= render partial: "shared/user_permissions", locals: { user_object: User.new } %>
-
-    <%= f.submit "Grant permissions to all users", :class => 'btn btn-success' %>
-  <% end %>
-</div>
+<% content_for :title, "Grant access to all users" %>
+
+<%= form_for @bulk_grant_permission_set do |f| %>
+  <%= render "govuk_publishing_components/components/select", {
+    id: "application_id",
+    label: "Application",
+    name: "application_id",
+    options: Doorkeeper::Application.all.map { |application| { text: application.name, value: application.id } }
```

Comment:
[In the shared/user_permissions partial the list of applications appears to be based on those that the user has access to](https://github.com/alphagov/signon/blob/41505143e821d5e5a64e7daa9ad9776ea8b3654d/app/views/shared/_user_permissions.html.erb#L15). Do we need to do something similar here? Or does the partial end up returning all applications because we're passing in `User.new`?

I see that the partial also differentiates between retired and non-retired applications. Was it a conscious decision to remove that distinction?

Should the applications be sorted alphabetically? Or is there some default sort order that means they're alphabetical?

Reply:
Thanks @chrislo - I've been through the extra commits and it all looks OK to me. I'll get this approved now.